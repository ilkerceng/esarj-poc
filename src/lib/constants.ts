import { AccountType, Status } from '../api/generated/model';

export const EMAIL_REGEX = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;

export const accountTypes = [
  { id: AccountType.Personal, label: 'Personal' },
  { id: AccountType.Corporate, label: 'Corporate' },
];

export const statuses = [
  { id: Status.Active, label: 'Active' },
  { id: Status.Passive, label: 'Passive' },
];

export const enumsData = {
  accountTypes,
  statuses,
};
