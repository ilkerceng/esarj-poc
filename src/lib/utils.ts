const leadingZerosPattern = '[^0]+';

export const splitWithLeadingPattern = (
  value: string,
  pattern = leadingZerosPattern,
): { leadingStr: string; otherPart: string } | undefined => {
  if (value) {
    const idStr = value.toString();
    const firstNonZeroNumIndex = value.toString().search(pattern);
    const leadingStr = idStr.substring(0, firstNonZeroNumIndex);
    const otherPart = idStr.substring(firstNonZeroNumIndex);

    return { leadingStr, otherPart };
  }
};

export function sum(a: number, b: number) {
  return a + b;
}
