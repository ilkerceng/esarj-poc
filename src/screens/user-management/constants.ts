import { BankOutlined, UserOutlined } from '@ant-design/icons';
import { AccountType } from '../../api/generated/model';

export const ACCOUNT_TYPE_MAP = {
  [AccountType.Personal]: UserOutlined,
  [AccountType.Corporate]: BankOutlined,
};
