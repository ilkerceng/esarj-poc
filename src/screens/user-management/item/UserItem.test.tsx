import renderer from 'react-test-renderer';
import { UserItem } from './UserItem';
import { accountTypes, statuses } from '../../../lib/constants';

//TODO: Tabs gives errors addEventListener error. https://github.com/ant-design/ant-design/issues/13620
jest.mock('rc-tabs/lib/Tabs', () => ({children}: any) => children);

test('new user renders correctly', () => {
  const tree = renderer
    .create(
      <UserItem
        id={0}
        isNewItem={false}
        accountTypes={accountTypes}
        statuses={statuses}
        onClose={() => {}}
        userData={undefined}
        isLoadingItem={false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// test('classNames set correctly', () => {
//   const { container } = render(<UserStatusBadge status={Status.Passive} />);
//   expect(container.firstChild).toHaveClass('');
// });
