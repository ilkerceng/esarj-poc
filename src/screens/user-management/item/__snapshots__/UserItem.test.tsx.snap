// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new user renders correctly 1`] = `
<div
  className="ant-card ant-card-bordered css-dev-only-do-not-override-2q8sxy"
  style={{}}
>
  <div
    className="ant-card-head"
    style={{}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        New User
      </div>
      <div
        className="ant-card-extra"
      >
        <div
          className="in"
        >
          <button
            className="ant-btn css-dev-only-do-not-override-2q8sxy ant-btn-circle ant-btn-default ant-btn-icon-only"
            color="#aaa"
            disabled={false}
            onClick={[Function]}
            style={{}}
            type="button"
          >
            <span
              className="ant-btn-icon"
              style={{}}
            >
              <span
                aria-label="close"
                className="anticon anticon-close"
                color="red"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close"
                  fill="currentColor"
                  fillRule="evenodd"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={{}}
  >
    <form
      className="ant-form ant-form-vertical css-dev-only-do-not-override-2q8sxy w-full"
      onReset={[Function]}
      onSubmit={[Function]}
      style={{}}
    >
      <div
        className="ant-divider css-dev-only-do-not-override-2q8sxy ant-divider-horizontal"
        role="separator"
        style={{}}
      />
      <div
        className="ant-row css-dev-only-do-not-override-2q8sxy"
        style={
          {
            "marginLeft": -8,
            "marginRight": -8,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="ant-col ant-col-12 css-dev-only-do-not-override-2q8sxy"
          style={
            {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <button
            className="ant-btn css-dev-only-do-not-override-2q8sxy ant-btn-primary"
            disabled={false}
            onClick={[Function]}
            style={{}}
            type="button"
          >
            <span
              className="ant-btn-icon"
              style={{}}
            >
              <span
                aria-label="edit"
                className="anticon anticon-edit"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="edit"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
                  />
                </svg>
              </span>
            </span>
            <span>
              Edit
            </span>
          </button>
        </div>
        <div
          className="ant-col ant-col-12 css-dev-only-do-not-override-2q8sxy"
          style={
            {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <div
            className="ant-form-item flex justify-end css-dev-only-do-not-override-2q8sxy"
          >
            <div
              className="ant-row ant-form-item-row css-dev-only-do-not-override-2q8sxy"
              style={
                {
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="ant-col ant-form-item-control css-dev-only-do-not-override-2q8sxy"
                style={{}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <button
                      className="ant-btn css-dev-only-do-not-override-2q8sxy ant-btn-primary"
                      disabled={true}
                      onClick={[Function]}
                      style={{}}
                      type="submit"
                    >
                      <span>
                        Submit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
